File: LeetCode/Easy/1837-sum-of-digits-in-base-k/1837-sum-of-digits-in-base-k.java
class Solution {
    public int sumBase(int n, int k) {
        int ret = 0;
        int i = 7;
        
        while (n < Math.pow(k, i))
            i--;

        while (i > 0){
            int temp = (int) Math.pow(k, i);
            ret += n / temp;
            n -= n / temp * temp;
            i--;
        }
        
        return ret + n;
    }
}

File: LeetCode/Easy/1837-sum-of-digits-in-base-k/1837-sum-of-digits-in-base-k.py
class Solution:
    def sumBase(self, n: int, k: int) -> int:
        ret = 0
        i = 7
        
        while n < k ** i:
            i -= 1

        while i:
            ret += n // k ** i
            n -= n // (k ** i) * (k ** i)
            i -= 1

        return ret + n